sentry/src/main/java/io/sentry/hints/Hints.java

```diff
@@ -1,29 +1,108 @@
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 public final class Hints {
 
   private final @NotNull Map<String, Object> internalStorage = new HashMap<String, Object>();
+  private final @NotNull List<Attachment> attachments = new CopyOnWriteArrayList<>();
+  private @Nullable Attachment screenshot = null;
 
-  public void set(@NotNull String hintType, @Nullable Object hint) {
-    internalStorage.put(hintType, hint);
+  public static @NotNull Hints withAttachment(@Nullable Attachment attachment) {
+    @NotNull final Hints hints = new Hints();
+    hints.addAttachment(attachment);
+    return hints;
   }
 
-  public @Nullable Object get(@NotNull String hintType) {
-    return internalStorage.get(hintType);
+  public static @NotNull Hints withAttachments(@Nullable List<Attachment> attachments) {
+    @NotNull final Hints hints = new Hints();
+    hints.addAttachments(attachments);
+    return hints;
   }
 
-  // TODO maybe not public
-  public void remove(@NotNull String hintType) {
-    internalStorage.remove(hintType);
+  public Hints() {
+    primitiveMappings = new HashMap<>();
+    primitiveMappings.put("boolean", Boolean.class);
+    primitiveMappings.put("char", Character.class);
+    primitiveMappings.put("byte", Byte.class);
+    primitiveMappings.put("short", Short.class);
+    primitiveMappings.put("int", Integer.class);
+    primitiveMappings.put("long", Long.class);
+    primitiveMappings.put("float", Float.class);
+    primitiveMappings.put("double", Double.class);
   }
 
-  // TODO addAttachment(one)
-  // TODO getAttachments(): List
-  // TODO setAttachments(list)
-  // TODO clearAttachments()
+  public void set(@NotNull String name, @Nullable Object hint) {
+    internalStorage.put(name, hint);
+  }
+
+  public @Nullable Object get(@NotNull String name) {
+    return internalStorage.get(name);
+  }
+
+  @SuppressWarnings("unchecked")
+  public <T extends Object> @Nullable T getAs(@NotNull String name, @NotNull Class<T> clazz) {
+    Object hintValue = internalStorage.get(name);
+
+    if (clazz.isInstance(hintValue)) {
+      return (T) hintValue;
+    } else if (isCastablePrimitive(hintValue, clazz)) {
+      return (T) hintValue;
+    } else {
+      return null;
+    }
+  }
+
+  public void remove(@NotNull String name) {
+    internalStorage.remove(name);
+  }
+
+  public void addAttachment(@Nullable Attachment attachment) {
+    if (attachment != null) {
+      attachments.add(attachment);
+    }
+  }
+
+  public void addAttachments(@Nullable List<Attachment> attachments) {
+    if (attachments != null) {
+      this.attachments.addAll(attachments);
+    }
+  }
+
+  public @NotNull List<Attachment> getAttachments() {
+    return new CopyOnWriteArrayList<>(attachments);
+  }
+
+  public void replaceAttachments(@Nullable List<Attachment> attachments) {
+    clear();
+    addAttachments(attachments);
+  }
+
+  public void clear() {
+    attachments.clear();
```

Comment:
`h`: Shouldn't this also clear the screenshot and the internal storage?

Reply:
Ah, forgot to rename.